### PR TITLE
finished with lab, going to work on Read me next

### PR DESCRIPTION
**Pull request by devonhackley**

### CODE REVIEW

**Comment by devonhackley** [comment]
###Lab: Stack and Queue Implementation

- [x] Node
- [x] Stack
      -[x] push
      -[x] pop
      -[x] peek
- [x] Queue
      -[x] enqueue
      -[x] dequeue
      -[x] peek
- [x] README

File: code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        this.front = null;
+        this.back = null;
+        this.size = 0;
+    }
+
+    public void enqueue(int value){
+        try
+        {
+            Node node = new Node(value);
+            node.next = null;
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java]: You set every new node's next to null in your constructor so I think this line is redundant

File: code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            }
+
+        } catch(IllegalArgumentException ex){
+            System.out.println("You either didn't pass anything in or you didn't pass in an Integer");
+        }
+    }
+
+    public int dequeue(){
+        if(this.front == null){
+            return 0;
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java]: How does the user know that 0 means that the Queue is empty? What is 0 is the value of the Node being dequeued?

File: code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            Node temp = this.front.next;
+            this.front = temp;
+            this.size--;
+            return curr.value;
+        }
+    }
+
+    public int peek(){
+        if(this.front == null){
+            return 0;
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java]: Same comment as above

File: code-challenges/401challenges/src/test/java/challenges/QueueTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        libraryForQueue.enqueue(4);
+        libraryForQueue.enqueue(5);
+        libraryForQueue.enqueue(6);
+
+        int peekVal = libraryForQueue.peek();
+        assertEquals(3, peekVal);
+        assertEquals(4, libraryForQueue.getSize());
+    }
+
+
```
Comment (evanslaton) [inline code-challenges/401challenges/src/test/java/challenges/QueueTest.java]: Extra empty line

File: code-challenges/401challenges/src/test/java/challenges/StackTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+        libraryToTest.push(3);
+        libraryToTest.push(4);
+        libraryToTest.push(5);
+
+        int peekVal = libraryToTest.peek();
+        assertEquals(5, peekVal);
+        assertEquals(3, libraryToTest.getLength());
+    }
+
```
Comment (evanslaton) [inline code-challenges/401challenges/src/test/java/challenges/StackTest.java]: Extra empty lines

File: code-challenges/401challenges/src/test/java/challenges/StackTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    @Test public void testStackPush(){
+        Stack libraryToTest = new Stack();
+
+        libraryToTest.push(3);
+        assertEquals(1, libraryToTest.getLength());
+        assertEquals(3, libraryToTest.getTop().value);
+    }
+    @Test public void testStackPushMultiple(){
```
Comment (evanslaton) [inline code-challenges/401challenges/src/test/java/challenges/StackTest.java]: No space before this test

File: code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    public Queue(){
+        this.front = null;
+        this.back = null;
+        this.size = 0;
+    }
+
+    public void enqueue(int value){
+        try
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java]: `try {`

**Pull request merged**